zcash_client_sqlite/src/wallet/init.rs

```diff
@@ -452,32 +453,36 @@ mod tests {
                            THEN 0
                          ELSE 1
                        END AS memo_present
-                FROM   sapling_received_notes
+                FROM sapling_received_notes
+                JOIN transactions
+                     ON transactions.id_tx = sapling_received_notes.tx
                 UNION
                 SELECT utxos.received_by_account     AS account_id,
-                       transactions.id_tx            AS id_tx,
+                       utxos.height                  AS block,
+                       utxos.prevout_txid            AS txid,
                        0                             AS pool,
                        utxos.value_zat               AS value,
                        0                             AS is_change,
                        1                             AS received_count,
                        0                             AS memo_present
                 FROM utxos
-                JOIN transactions
-                     ON transactions.txid = utxos.prevout_txid
                 UNION
                 SELECT sapling_received_notes.account        AS account_id,
-                       sapling_received_notes.spent          AS id_tx,
-                       2                             AS pool,
+                       transactions.block                    AS block,
+                       transactions.txid                     AS txid,
+                       2                                     AS pool,
                        -sapling_received_notes.value         AS value,
                        0                             AS is_change,
                        0                             AS received_count,
                        0                             AS memo_present
-                FROM   sapling_received_notes
+                FROM sapling_received_notes
+                JOIN transactions
+                     ON transactions.id_tx = sapling_received_notes.spent
                 WHERE  sapling_received_notes.spent IS NOT NULL
```

Comment:
I think this `WHERE` is now unnecessary, given that `JOIN` is an inner join? We could make the latter explicit here to clarify we only want spent notes:
```suggestion
                INNER JOIN transactions
                     ON transactions.id_tx = sapling_received_notes.spent
```